cairo-libs/packages/protobuf/src/types/message.cairo

```diff
@@ -96,6 +96,15 @@ pub impl EncodeContextImpl of EncodeContextTrait {
         }
         self.buffer.append(@context2.buffer);
     }
+
+    fn encode_with_length<T, +ProtoMessage<T>, +Default<T>, +Drop<T>, +PartialEq<T>>(
+        ref self: EncodeContext, value: @T,
+    ) {
+        let mut context2 = Self::new();
+        value.encode_raw(ref context2);
+        context2.buffer.len().encode_raw(ref self);
+        self.buffer.append(@context2.buffer);
```

Comment:
Any reason for naming it `context2`?
```suggestion
        let mut context = Self::new();
        value.encode_raw(ref context);
        context.buffer.len().encode_raw(ref self);
        self.buffer.append(@context.buffer);
```

Reply:
good catch ! I was copy-pasting 😄

Reply:
Ah. I just realized that I was using `context2` to signify it is a second context or sub-context -- `self` being the first context or parent-context.